interfaces/apparmor/backend.go

```diff
@@ -446,6 +446,29 @@ func addUpdateNSProfile(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	}
 }
 
+func downgradeConfinement() bool {
+	kver := osutil.KernelVersion()
+	switch {
+	case release.DistroLike("opensuse-tumbleweed"):
+		if cmp, _ := strutil.VersionCompare(kver, "4.16"); cmp >= 0 {
+			// As a special exception, for openSUSE Tumbleweed which ships Linux
+			// 4.16, do not downgrade the confinement template.
+			return false
+		}
+	case release.DistroLike("arch"):
+		if !strings.HasSuffix(kver, "-hardened") {
+			// how did we get here anyway?
+			return true
+		}
+		if cmp, _ := strutil.VersionCompare(kver, "4.17.4"); cmp >= 0 {
```

Comment:
Hm, won't we need to do the version check first and then the harden check? If the version is `4.17.4` then the `downgradeConfinement` helper here would return "true" because it first checks for !HasSuffix("-hardened") and that is false so it returns early. This also indicates a test is missing :)